internal/api/oidc/client.go

```diff
@@ -287,6 +304,18 @@ func (o *OPStorage) assertUserMetaData(ctx context.Context, userID string) (map[
 	return userMetaData, nil
 }
 
+func (o *OPStorage) assertUserResourceOwner(ctx context.Context, userID string) (map[string]string, error) {
+	resourceOwner, err := o.repo.OrgByUserID(ctx, userID)
+	if err != nil {
+		return nil, err
+	}
+	return map[string]string{
+		"id":             resourceOwner.AggregateID,
+		"name":           resourceOwner.Name,
+		"primary_domain": resourceOwner.PrimaryDomain,
+	}, nil
```

Comment:
IMO appending a separate claim as map is confusing... there now might be two claims where the primary domain is listed: urn:zitadel:iam:user:resourceowner and urn:zitadel:iam:org:domain:primary
wouldn't it make sense to have three simple claims (resourceowner/org, org:domain:primary and org:name) which would be filled whenever one of the scopes (resourceowner, org:domain:primary or org:id) is requested?